src/readinglistbar.cpp

```diff
@@ -60,13 +65,43 @@ void ReadingListBar::setupList()
     }
 }
 
-void ReadingListBar::on_itemActivated(QListWidgetItem* item)
+void ReadingListBar::on_itemActivated(QListWidgetItem* item) // Handles double click and enter key
+{   
+    lastDoubleClickTime = QDateTime::currentDateTime();
+    openUrl(item, true);
+}
+
+void ReadingListBar::on_itemClicked(QListWidgetItem* item, Qt::MouseButtons buttons) // Handles single and middle click
+{   
+    if(lastDoubleClickTime.isValid()) {
+        QDateTime thisClickTime = QDateTime::currentDateTime();
+        // Give time for a double click be handled
+        QEventLoop evtLoop;
+        int doubleClickInterval = QApplication::doubleClickInterval();
+        QTimer::singleShot(doubleClickInterval, &evtLoop, &QEventLoop::quit);
+        evtLoop.exec();
```

Comment:
Isn't this merely a more convoluted way of doing `QThread::msleep(QApplication::doubleClickInterval())`;

Reply:
Thanks for bringing this up because it is something I wanted to discuss. Essentially, the difference is that the singleShot alternative allows the event loop to process other events in the event queue. On the other hand the code that you provided makes the thread block while sleeping. 

I am not sure which is more sufficient in this case, but I initially chose the first one to allow for better execution. LMK if you have any doubts.

Reply:
Additionally, after testing this, it seems to me that the application freezes for a small amount of time when double clicking on a reading list bookmark.

Reply:
So the answer to my question is 'no' (because of the difference in the blocking behaviour).

Now, supposing that `ReadingListBar::on_itemSingleClicked()` is nevertheless introduced you can rewrite this code in a more comprehensible way as follows:

```
QTimer::singleShot(doubleClickInterval, [=]() {
    const qint64 timePassed = lastDoubleClickTime.msecsTo(thisClickTime);
    if ( timePassed <= doubleClickInterval ) {
        on_itemSingleClicked(item, buttons);
    }
});
```

Reply:
After testing, using the code below, this does not work as intended. 
```
QDateTime thisClickTime = QDateTime::currentDateTime();
int doubleClickInterval = QApplication::doubleClickInterval();
QTimer::singleShot(doubleClickInterval, [=]() {
    const qint64 timePassed = lastDoubleClickTime.msecsTo(thisClickTime);
    if (timePassed <= doubleClickInterval) { return; }
});
QEventLoop().exec();
```

I have left it commented out, in my last commit, if you want to check.

Reply:
Your code differs from my proposal significantly (the `return` statement has no effect since it returns from the lambda which is about to complete on the next line anyway). However, mine contained a bug too. It should have been as follows (note the corrected condition in the `if` statement):

```c++
int doubleClickInterval = QApplication::doubleClickInterval();
QTimer::singleShot(doubleClickInterval, [=]() {
    const qint64 timePassed = lastDoubleClickTime.msecsTo(thisClickTime);
     if (timePassed > doubleClickInterval) {
         itemSingleClicked(item);
     }
});
```

Reply:
My bad, I misunderstood that you were intending to do something different. This works perfect.

Also, sorry for the very late feedback. Give me a bit of time to get back to you on the rest as well.